docs/features/traffic-management/istio.md

```diff
@@ -1,3 +1,88 @@
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: rollout-vsvc
+spec:
+  gateways:
+    - istio-rollout-gateway
+  hosts:
+    - istio-rollout.dev.argoproj.io
+  http:
+    - name: primary
+      route:
+        - destination:
+            host: stable-svc
+          weight: 100
+        - destination:
+            host: canary-svc
+          weight: 0
+```
+
+When the above Rollout progresses through its steps, the controller changes Virtual Service's `stable-svc`'s weight to 95 and `canary-svc`'s to 5, wait 5 minutes, and then scales up the canary ReplicaSet to a full replica count. Once it is entirely healthy, the controller changes `stable-svc`'s selector to point at the canary ReplicaSet and switch the weight back to 100 to `stable-svc` and 0 to `canary-svc`.
+
+Note: The Rollout does not make any other assumptions about the fields within the Virtual Service or the Istio mesh. The user could specify additional configurations for the virtual service like URI rewrite rules on the primary route or any other route if desired. The user can also create specific destination rules for each of the services. 
+
+
+## Integrating with GitOps
+The above strategy introduces a problem for users practicing GitOps. The Rollout requires the user-defined Virtual Service to define an HTTP route with both destinations hosts. However, Istio requires routes with multiple destinations to assign a weight to each destination. Since the Argo Rollout controller modifies these Virtual Service's weights as a Rollout progresses through its steps, the Virtual Service becomes out of sync with the Git version.
+Additionally, if a GitOps tool does an apply after the Argo Rollouts controller changes the Virtual Service's weight, the apply would revert the weight to the percentage stored in the Git repo. At best, the user can specify the desired weight of 100% to the stable service and 0% to the canary service. In this case, the Virtual Service is synced with the Git repo when the Rollout completed all the steps. 
+
+Argo CD has an [open issue here](https://github.com/argoproj/argo-cd/issues/2913) to address this problem. The solution is that Argo CD supports adding an annotation called `argo-cd.argoproj.io/respect-existing-fields` to the Virtual Service. This annotation indicates that Argo CD should respect the current weights listed in this virtual service and rely on the controller to honor the values of the weights within the CR.
```

Comment:
Let's not get into any specifics just yet. Please reword that a proposed solution is to use an argo-cd specific annotation which prevents overwriting the weights during a sync operation. Feel free to rephrase.